packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -202,10 +202,23 @@ class DataSourceWithBackend<
     headers[PluginRequestHeaders.PluginID] = Array.from(pluginIDs).join(', ');
     headers[PluginRequestHeaders.DatasourceUID] = Array.from(dsUIDs).join(', ');
 
-    let url = '/api/ds/query';
+    let queryStrObj: Record<string, string> = {};
+
+    let url = '/api/ds/query?ds_type=' + this.type;
+
     if (hasExpr) {
       headers[PluginRequestHeaders.FromExpression] = 'true';
-      url += '?expression=true';
+      queryStrObj.expression = 'true';
```

Comment:
```suggestion
      url += '&expression=true';
```

Reply:
then get rid of URLSearchParams... we don't have anything with crazy escaping etc

Reply:
Uh why tho? it's already bundled up in the queryStr

Reply:
I guess we could avoid using the `URLSearchParams` altogether since we've always got a ds_type

Reply:
Sorry, github didn't serve me your second commend right away. Fixed